### ISSUE TITLE
Add documentation style checker 

### ISSUE THREAD

**Issue by bitwiseman** [josdejong/mathjs#394]
As this project grows, having clear and consistent documentation will become more important.  Docs are already generated from comments.

For example: 
* Spellchecking (no: "dimesnion")
* List of approved/rejected phrases (no: "one dimensional", yes: "one-dimensional")
* Check supported argument types match docs

**Comment by josdejong** [comment]
Yeah, we could extend the `docgenerator` used to generate docs or put an extra tool to test for more cases like typos and specific phrases. I don't know whether this would be complicated to build or whether we would really gain much from this. My IDE contains a spell checker so I mostly see typos straight away when writing code. 

Maybe we could start simple by putting a list with phrases in [CONTRIBUTING.md](https://github.com/josdejong/mathjs/blob/master/CONTRIBUTING.md)? Do you have any suggestions?

**Comment by devanp92** [comment]
Also, it would be important to add a consistent structure to each doc. Such as: 

* name
* description
* syntax
* example
* etc.
